### PR TITLE
Add Github Actions workflow that builds a PyPI package (and deploys it on PyPI if Github release) + simple test with barebones conda environment file

### PR DESCRIPTION
**Pull request by valeriupredoi**
I've added a convenient PyPI package builder Github Action that builds it with the latest supported Python (nominally Python=3.12 since #19 #20 and the final #21 ), and the upload should be automated via a token that we'd have to upload to PyPI. I am also adding a barebones `pyproject.toml` file to be used with the pkg build; the build seems to [run well](https://github.com/cedadev/MyProxyClient/actions/runs/6853071338/job/18633058138?pr=22) - the upload (in case of release) can only happen if we upload the token to PyPI - do any of you gets want to do that?

EDIT: waiting for good man @philipkershaw to make me an owner of the PyPI repo, I have also added a barebones unit test + barebones conda env file to have the test run on multiple Python versions

### CODE REVIEW

File: .github/workflows/build-and-deploy-on-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      #- name: Publish distribution 📦 to Test PyPI
+      #  uses: pypa/gh-action-pypi-publish@master
+      #  with:
+      #    password: ${{ secrets.test_pypi_password }}
+      #    repository_url: https://test.pypi.org/legacy/
+      - name: Publish distribution 📦 to PyPI
+        if: startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
```
Comment (bouweandela) [inline .github/workflows/build-and-deploy-on-pypi.yml:46]: You may want to have a look at trusted publishers instead of using a username and password, is the new thing: https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/

Comment (valeriupredoi) [reply .github/workflows/build-and-deploy-on-pypi.yml:46]: ah evolution! Great pointer @bouweandela - cheers! Do I take it I can ask you for reviews in the future? Feels very lonely around 'ere 😁

Comment (bouweandela) [reply .github/workflows/build-and-deploy-on-pypi.yml:46]: Yes, I would be happy to help out :beers:

**Comment by valeriupredoi** [comment]
@bouweandela I have not forgotten about this, it's just that we need access to PyPI, via trusted publishers etc (am talking to Phil over Slack) :beer:

**Comment by valeriupredoi** [comment]
@bouweandela do you have a bit of time give this a revew please, we can immediately deploy on PyPI and make a new release on conda-forge, as soon as Phil allows me ownership of the PyPI project, so we can publish there. This now has a tiny test and a GA workflow that installs MyProxyClient locally and runs the tiny test; just so we can have something (than nothing) in the way of testing :beer:

File: tests/unit/test_simple.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        'ssl_verification', 'store', 'writeProxyFile'
+    ]
+    actual_members = dir(MyProxyClient)
+    for member in expected_members:
+        assert member in actual_members
+
+
+def test_simple():
+    """Test a simple instance of MyProxyClient."""
+    hostname = "www.google.com"
```
Comment (bouweandela) [inline tests/unit/test_simple.py:21]: It would be a bit nicer to use an example service, e.g. example.com instead of a random actual website

File: .github/workflows/build-and-deploy-on-pypi.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          .
+      #- name: Publish distribution 📦 to Test PyPI
+      #  uses: pypa/gh-action-pypi-publish@master
+      #  with:
+      #    password: ${{ secrets.test_pypi_password }}
+      #    repository_url: https://test.pypi.org/legacy/
+      - name: Publish distribution 📦 to PyPI
+        if: startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+      # this is the old token and password way;
```
Comment (bouweandela) [inline .github/workflows/build-and-deploy-on-pypi.yml:56]: Could you clean up by removing all the commented out stuff?

Comment (valeriupredoi) [reply .github/workflows/build-and-deploy-on-pypi.yml]: done!

File: tests/unit/test_simple.py (hunk 2 above)
Comment (valeriupredoi) [reply tests/unit/test_simple.py]: imaginary domain now in, bud

**Comment by bouweandela** [comment]
Nice that the test upload succeeded!

I noticed a warning in the upload, could you add 
```python
long_description_content_type='text/markdown',
```
below this line:
https://github.com/cedadev/MyProxyClient/blob/25d8710e31a330f18dac58809d7101131d828636/setup.py#L46 
?

File: .github/workflows/build-and-deploy-on-pypi.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - add_pypi_pkg_build_GA
+
+jobs:
+  build-n-publish:
+    name: Build and publish MyProxyClient on PyPi
+    runs-on: ubuntu-latest
+    # may also need this for publishing?
+    environment:
+      name: pypi
+      url: https://pypi.org/project/MyProxyClient/
```
Comment (bouweandela) [inline .github/workflows/build-and-deploy-on-pypi.yml:19]: Why is this needed?

File: .github/workflows/build-and-deploy-on-pypi.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          python -m
+          pep517.build
+          --source
+          --binary
+          --out-dir dist/
+          .
+      - name: Publish distribution 📦 to Test PyP
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
```
Comment (bouweandela) [inline .github/workflows/build-and-deploy-on-pypi.yml:52]: Could you remove this once you're done testing please?

Comment (valeriupredoi) [reply .github/workflows/build-and-deploy-on-pypi.yml:52]: right now, before merging

File: .github/workflows/build-and-deploy-on-pypi.yml (hunk 4 above)
Comment (valeriupredoi) [reply .github/workflows/build-and-deploy-on-pypi.yml:19]: the instructions at that site in the comment say we should add that - not sure of it myself, but I thought I'd follow instructions ad literam

**Review by bouweandela — APPROVED** [review]

**Comment by valeriupredoi** [comment]
very many thanks @bouweandela :beer:
